march_gait_selection/src/march_gait_selection/gait_selection.py

```diff
@@ -72,6 +71,8 @@ def load_gaits(self):
             loaded_gait = Gait.from_file(gait, self.gait_directory, self.robot, self._gait_version_map)
             self.loaded_gaits.append(loaded_gait)
 
+        self.balance_gait.default_walk = next((gait for gait in self.loaded_gaits if gait.gait_name == 'walk'), None)
```

Comment:
```suggestion
        self.balance_gait.default_walk = self['walk']
```
This uses the __getitem__ method ;)